include/swift/AST/DiagnosticsParse.def

```diff
@@ -150,7 +150,8 @@ ERROR(lex_invalid_identifier_start_character,none,
 ERROR(lex_expected_digit_in_fp_exponent,none,
        "expected a digit in floating point exponent", ())
 ERROR(lex_expected_digit_in_int_literal,none,
-       "expected a digit after integer literal prefix", ())
+      "expected %select{a binary digit (0-1)|an octal digit (0-7)|a digit|a hex"
```

Comment:
Another nit: since the other messages (see immediately below) also spell it out, probably more consistent to say "hexadecimal digit" instead of "hex digit"; also, could we put a space in `(0-7,A-F)` after the comma?